Casks/a/adrive.rb

```diff
@@ -10,8 +10,8 @@
   homepage "https://www.aliyundrive.com/"
 
   livecheck do
-    url :url
-    strategy :extract_plist
+    url "https://www.aliyundrive.com/download"
+    regex(/aDrive-(\d+(?:\.\d+)+)\.dmg/i)
```

Comment:
```suggestion
    regex(/href=.*?aDrive[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
```

This follows our regular regex standards.

Reply:
- The download link is not a href on https://www.aliyundrive.com/download , `href=.*?` cannot be matched

- For the `[._-]v?` part, if the download link is not in same format, the `url` will not work.

https://github.com/Homebrew/homebrew-cask/blob/9463df529bbf9cd1e0000a6a13b3dff10968fc9b/Casks/a/adrive.rb#L5

Reply:
`curl` returns ` app_mac_download_link: 'https://cdn.aliyundrive.net/downloads/apps/desktop/aDrive-4.11.0.dmg',`. So we should drop the `href=.*?` but keep `[._-]v?` here. I've verified that updated regex works.